bundles/org.openhab.core.transform/src/main/java/org/openhab/core/transform/TransformationConfiguration.java

```diff
@@ -28,21 +28,24 @@ public class TransformationConfiguration implements Identifiable<String> {
     private final String uid;
     private final String label;
     private final String type;
+    private final @Nullable String context; // for backward deserialization compatibility
     private final @Nullable String language;
     private final String content;
 
     /**
      * @param uid the configuration UID. The format is config:&lt;type&gt;:&lt;name&gt;[:&lt;locale&gt;]. For backward
      *            compatibility also filenames are allowed.
      * @param type the type of the configuration (file extension for file-based providers)
+     * @param context context (e.g. script type or blockly blocks) of the configuration as JSON string
```

Comment:
I'm not a big fan of expecting JSON format in Java String parameters. It especially is totally unclear what kind of values are expected to be passed here and how they are potentially used. Why isn't the transformation type not enough? It is supposed to describe the transfomation and with it the expected configuration of it.

Reply:
There are two reasons: 

- For good UI integration we need to give a hint to the UI which editor to use. For "map" or "scale" we can also integrate that into the UI but e.g. the "script" transformation needs different editors for blocky, JS or Python and there is no way to auto-detect the correct scripting language. 
- For blocky "scripts" we need to store additional data (the blockly "code") to edit them later. 

If you have a good idea how to store arbitrary data other than an encoded JSON, I'm happy to implement that.